### PR TITLE
Node hover options

### PR DESCRIPTION
**Pull request by JethroWarnett**
A buggy implementation of a node hover option. When hovering a node a smiley face appears, clicking this smiley will add a child node to the hovered node. The code is buggy and it will be fixed in a refactor.

### CODE REVIEW

File: src/app/components/fairChain/fairChain.component.ts [hunk 1]
```diff
@@ -81,12 +99,40 @@ export class FairChainComponent implements OnInit {
...
+      this.genHexString(12);
+  }
+
+  private genHexString(len) {
+    const hex = '0123456789abcdef';
+    let output = '';
+    for (let i = 0; i < len; ++i) {
+        output += hex.charAt(Math.floor(Math.random() * hex.length));
+    }
+    return output;
```
Comment (GattoPianista) [inline src/app/components/fairChain/fairChain.component.ts:127]: Is it not a risk if we make it random?

File: src/app/components/fairChain/fairChain.component.ts [hunk 2]
```diff
@@ -433,6 +485,65 @@ export class FairChainComponent implements OnInit {
...
   }
 
+  makeMouseHoveringNodeSubscription(boundingBox: DOMBoundingBox) {
+    this.nodeHoverSubscription = fromEvent(document, 'mousemove').pipe(
+      filter((pointer: MouseEvent) => {
+        if (!(boundingBox.left <= pointer.clientX
+          && pointer.clientX <= boundingBox.right 
+          && boundingBox.bottom <= pointer.clientY
+          && pointer.clientY <= boundingBox.top))
+          return true;
```
Comment (GattoPianista) [inline src/app/components/fairChain/fairChain.component.ts:495]: Maybe a method for this would make clearer what is going on

**Review by GattoPianista — APPROVED** [review]
Nice job! I added some comments, but they are just work that you would have done in any case

**Review by LYCChiang — APPROVED** [review]

**Review by Korab-Bejta — APPROVED** [review]

**Review by mdeliso97 — APPROVED** [review]
(Sorry for the delay)
The code seems fine.
THe implementation itself is really good and confortable to use.
I found one bug:
- When you change label to the edges dynamically and after change the color of a node, the node changes color and also label which is weird.
The rest works perfectly, well done. After this fix the branch can be merged into dev

**Comment by LYCChiang** [comment]
When you change the color, the label disappears

**Pull request closed without merging**